pkg/controller/storage/mem/storage_binding.go

```diff
@@ -38,35 +38,62 @@ func newMemStorageBinding() *memStorageBinding {
 }
 
 func (m *memStorageBinding) List() ([]*servicecatalog.Binding, error) {
-	ret := make([]*servicecatalog.Binding, len(m.bindings))
+	copy := make([]*servicecatalog.Binding, len(m.bindings))
 	i := 0
-	for _, val := range m.bindings {
-		ret[i] = val
+	for _, bin := range m.bindings {
+		copy[i] = &servicecatalog.Binding{}
+		if err := deepCopy(bin, copy[i]); err != nil {
+			return nil, err
+		}
 		i++
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Get(name string) (*servicecatalog.Binding, error) {
-	ret, ok := m.bindings[name]
+	binding, ok := m.bindings[name]
 	if !ok {
-		return nil, errNoSuchBinding
+		return nil, errNoSuchInstance
+	}
+	copy := &servicecatalog.Binding{}
+	if err := deepCopy(binding, &copy); err != nil {
+		return nil, err
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Create(bin *servicecatalog.Binding) (*servicecatalog.Binding, error) {
 	if _, err := m.Get(bin.Name); err == nil {
 		return nil, errBindingAlreadyExists
 	}
-	m.bindings[bin.Name] = bin
-	return bin, nil
+	copy1 := &servicecatalog.Binding{}
+	if err := deepCopy(bin, copy1); err != nil {
+		return nil, err
+	}
+	copy2 := &servicecatalog.Binding{}
+	if err := deepCopy(bin, copy2); err != nil {
+		return nil, err
+	}
+	m.bindings[bin.Name] = copy1
```

Comment:
totally agree in general that the returned object should have all of the updated data as seen by what's in the data store. But as long as you're not returning a pointer to our in-memory DB one, in this case I don't think it really matters that we return the same (original) pointer since it will have the right data.  

You don't have to change it - I just thought the extra copy() calls seemed unnecessary.